### DISCUSSION TITLE
FreeU for Fooocus

### DISCUSSION THREAD

**Discussion by kubilaykilinc** [lllyasviel/Fooocus#485] [Ideas]
Would it be possible to include FreeU in Fooocus? or can be added as an option.
https://chenyangsi.top/FreeU/

code : https://github.com/ChenyangSi/FreeU

This is something new though. I don't think it's fully seated yet. It has a positive contribution for SD 1.5 and 2.1. but I don't know for SDXL 1.0. I'll test it a bit on comfyUI. Still, I wanted to write this idea here.

**Comment by drdancm** [comment]
Sounds great.

**Comment by vesper8** [comment]
That indeed sounds most excellent! Does anyone know how to use it with Fooocus in the meantime?

**Comment by AkoZ** [comment]
MEMO to keep back things
i add back here the references found on their github (Chnyangsi): 
FreeU connected between Checkpoint and Ksampler with 4 parameters as B1, b2 and s1, s2
* [@murphylanga147](https://www.youtube.com/channel/UCX3CsqqI_MtPiNPreR16q7Q)
* the B values (around 1) control the style and image quality as the S control the quality/number (e.g. number of legs of an animal) of the objects on the image
* best results with values around and higher for S1 and with low values between 0-0.3 for S2.
* When the distance between S1 and S2 becomes smaller, more and more objects are generated.
* ... tiny details and textures are the S, and the larger blobs are B
[Results from the community](https://github.com/ChenyangSi/FreeU#results-from-the-community)
If you tried FreeU and want to share your results, let me know and we can put up the link here.

[SDXL](https://wandb.ai/nasirk24/UNET-FreeU-SDXL/reports/FreeU-SDXL-Optimal-Parameters--Vmlldzo1NDg4NTUw?accessToken=6745kr9rjd6e9yjevkr9bpd2lm6dpn6j00428gz5l60jrhl3gj4gubrz4aepupda) from [Nasir Khalid](https://wandb.ai/nasirk24):
b1 = 1.1 - b2 = 1.2 - s1 = 0.6 -  s2 = 0.4


* [comfyUI](https://twitter.com/bramvera/status/1706190498220884007) from [Abraham](https://twitter.com/bramvera)
* [ComfyUI-FreeU](https://www.youtube.com/watch?v=8XJH6uZjNzA&t=297s) (YouTube)
* [ComfyUI-FreeU](https://www.bilibili.com/video/BV1om4y1G7TX/) (中文)

[SD2.1](https://twitter.com/justindujardin/status/1706021278963179612) from [Justin DuJardin](https://twitter.com/justindujardin)
[SDXL](https://twitter.com/seb_cawai/status/1705948389874000374) from [Sebastian](https://twitter.com/seb_cawai)
[SDXL](https://twitter.com/tintwotin/status/1706318393312223346) from [tintwotin](https://twitter.com/tintwotin)
[Rerender](https://github.com/williamyang1991/Rerender_A_Video#freeu)
[Collaborative-Diffusion](https://github.com/ziqihuangg/Collaborative-Diffusion/tree/master/freeu)

**Comment by lllyasviel** [comment]
it is in developer debug tools after 2.1.28

**Reply by vicento** [reply]
thanks

**Comment by DavidAllenNeron** [comment]
How exactly should this be used?
Do I turn select NONE for Refiner if I use FreeU?

**Comment by MoonRide303** [comment]
You can find some examples illustrating how you can use FreeU in [MRE wiki](https://github.com/MoonRide303/Fooocus-MRE/wiki/FreeU). Results in original Fooocus for the same values might be a bit different, because I've updated FreeU node to v2 in Fooocus v2.1.701 (https://github.com/lllyasviel/Fooocus/pull/727).
